src/core/SetupApi.ts

```diff
@@ -10,12 +10,38 @@ import { pipeEvents } from './utils/internal/pipeEvents'
 import { toReadonlyArray } from './utils/internal/toReadonlyArray'
 import { Disposable } from './utils/internal/Disposable'
 
+export abstract class HandlersController {
```

Comment:
This may be over-architecturing on my part so I'm open to feedback. The idea is that the request handlers list management is kind of its own thing. Overriding `.use()` and `.reset()` methods in child classes also require them to re-implement additional logic, like the input validation, and it feels a bit redundant. 

@mattcosta7, what is your opinion on this?

Reply:
I don't mind this explicit abstract class.  Agreed it might be heavier than necessary, but it handles interface definition and provides a nice wrapper around the fixes 'array' storage for handlers, which I also kind of like, so I'm for it

Reply:
Thank you, @mattcosta7!